compiler/luci/pass/src/QuantizeDequantizeWeightsWithGPTQPass.cpp

```diff
@@ -0,0 +1,200 @@
+
+    if (not is_quantizable(node->filter()))
+      return;
+
+    auto weights = loco::must_cast<luci::CircleConst *>(node->filter());
+    auto new_weights = luci::clone(weights);
+    node->filter(new_weights);
+
+    fake_quantize(new_weights);
+  }
+
+  void visit(luci::CircleTransposeConv *node)
+  {
+    LOGGER(l);
+    INFO(l) << "QuantizeDequantizeWeightsWithGPTQPass visit node: " << node->name() << std::endl;
+
+    if (not is_quantizable(node->filter()))
+      return;
+
+    auto weights = loco::must_cast<luci::CircleConst *>(node->filter());
+    auto new_weights = luci::clone(weights);
+    node->filter(new_weights);
+
+    fake_quantize(new_weights);
+  }
+
+  void visit(luci::CircleFullyConnected *node)
+  {
+    LOGGER(l);
+    INFO(l) << "QuantizeDequantizeWeightsWithGPTQPass visit node: " << node->name() << std::endl;
+    if (not is_quantizable(node->weights()))
+      return;
+
+    auto weights = loco::must_cast<luci::CircleConst *>(node->weights());
+    auto new_weights = luci::clone(weights);
+    node->weights(new_weights);
+
+    auto hessian = (*_hessian_map)[node];
+
+    fake_quantize_with_gptq(new_weights, hessian);
+  }
+};
+
+} // namespace
+
+bool QuantizeDequantizeWeightsWithGPTQPass::run(loco::Graph *g)
+{
+  LOGGER(l);
+  INFO(l) << "QuantizeDequantizeWeightsWithGPTQ Start" << std::endl;
+
+  if (_ctx->input_model_dtype != loco::DataType::FLOAT32)
+    throw std::runtime_error("GPTQPass: Weights-only quantization supports float32 input only");
+
+  if (_ctx->output_model_dtype != loco::DataType::U8 &&
+      _ctx->output_model_dtype != loco::DataType::U4)
+  {
+    throw std::runtime_error("GPTQPass: GPTQ quantization supports uint4/uint8");
+  }
+
+  auto info_by_name = layer_info_map(g, _ctx->layers_info);
+
+  auto quantize_dtype = [&](const luci::CircleNode *node) {
+    auto iter = info_by_name.find(node->name());
+
+    // Return designated quantization dtype
+    if (iter != info_by_name.end())
+      return iter->second.dtype;
+
+    // Return default quantization dtype
+    return _ctx->output_model_dtype;
+  };
+
+  auto quantize_granularity = [&](const luci::CircleNode *node) {
+    auto iter = info_by_name.find(node->name());
+
+    // Return designated quantization granularity
+    if (iter != info_by_name.end())
+      return iter->second.granularity;
+
+    // Return default quantization granularity
+    return _ctx->granularity;
+  };
+
+  // Quantize weights
+  for (auto node : loco::active_nodes(loco::output_nodes(g)))
+  {
+    auto circle_node = loco::must_cast<luci::CircleNode *>(node);
+    QuantizeDequantizeWeightsWithGPTQ qw(_ctx->input_model_dtype, quantize_dtype(circle_node),
+                                         quantize_granularity(circle_node), _hessian_map);
+    circle_node->accept(&qw);
+  }
+
+  INFO(l) << "QuantizeDequantizeWeightsWithGPTQ End" << std::endl;
+  return false;
```

Comment:
why return `false`? do you know the meaning of true/false?

Reply:
I think `false` is returned because this pass is run only once. It might come from QuantizeWeights.cpp (existing code).

Reply:
Plz add the same comment so that we know the reason

Reply:
Thanks for the explanation, @jinevening. I got it now after checking about what return true/false means in the `Pass` process. Also, I've added the same comments as the existing code.